Climber PID magik

WAaron-500 opened the pull request.

WAaron-500 commented on src/main/java/frc/robot/util/TalonFXUtils.java:
```diff
@@ -0,0 +1,32 @@
+package frc.robot.util;
+
+public class TalonFXUtils {
+
+    public static final double TICKS_PER_REVOLUTION = 2048;
+
+    /**
+     * Converts ticks per 100ms to rotations per minute.
+     * @param ticks ticks per 100ms
+     * @return RPM
+     */
+    public static double ticksPer100msToRPM(double ticks) {
+        return ticks * 600 / TICKS_PER_REVOLUTION;
+    }
+
+    /**
+     * 
+     * @param rpm rotations per minute
+     * @return ticks per 100ms
+     */
+    public static double RPMToTicksPer100ms(double rpm) {
+        return rpm * TICKS_PER_REVOLUTION / 600;
+    }
+
+    public static double metersToTicks(double meters, double gearRatio, double wheelRadius) {
+        return meters * TICKS_PER_REVOLUTION * gearRatio / (2 * Math.PI * wheelRadius);
+    }
+
```
move this to package when it comes out

FellStinger replied:
I was just thinking we'd put it in a separate package within the repo, rather than the same package as all of our robot code, like putting it in frc.lib.util instead of frc.robot.util

WAaron-500 replied:
ye

hadley31 commented:
Here are some ideas I have:
1. Introduce a boolean method in climber which returns if pid is at setpoint.
2. Climber position PID commands - isFinished should return whether the controller is at the setpoint. Or possibly have a boolean option to immediately finish or wait.
3. Revert to individual climber subsystem modules (I can elaborate on this later)
4. Is the PID slots so that we can use different instances? Like have a pid for velocity and one for position? If so, we should use slot 1 for position

WAaron-500 commented:
Alright Nick I have Implemented 1 and 2 without problems, I think and in doing so I think I may have found something that could have done a real bad, but I think I fixed it... had to do with something not stopping when its supposed to.

NVM I FOUND IT TURNS OUT THERE IS A SUBTLE DIFFERENCE BETWEEN WHAT I WAS REFERING TO 4 being pushed soon
